lib/team_statistics.rb

```diff
@@ -0,0 +1,200 @@
+require_relative "game_data"
+require_relative "team_data"
+require_relative "game_team_data"
+require_relative "league_statistics"
+require "csv"
+
+class TeamStatistics < LeagueStatistics
+
+  def initialize
+    @team_info_by_id = Hash.new
+    @by_team_id_game_objects = []
+    @total_games_by_season = Hash.new{ |hash, key| hash[key] = 0 }
+    @total_wins_by_season = Hash.new{ |hash, key| hash[key] = 0 }
+    @win_percentage_by_season = Hash.new
+    @games_won_by_team_id = Hash.new{ |hash, key| hash[key] = 0 }
+    @goals_by_game = []
+    @team_name_by_id = Hash.new{}
+    @games_played_by_opponent = Hash.new{ |hash, key| hash[key] = 0 }
+    @games_won_by_opponent = Hash.new{ |hash, key| hash[key] = 0 }
+    @win_ratio_by_opponent = Hash.new
+  end
+
+  def all_teams
+    TeamData.create_objects
+  end
+
+  def all_games
+    GameData.create_objects
+  end
+
+  def all_game_teams
+    GameTeamData.create_objects
+  end
+
```

Comment:
Inheritance/Modules for these?

Reply:
I think this will Inherit from a Superclass of Futbol Data